### PR TITLE
[logo] Fix disabling logo in xml

### PR DESCRIPTION
**Pull request by kiryldz**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [x] Add example if relevant.
 - [x] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>[logo] Fix disabling logo in xml.</changelog>`.

### Summary of changes

Fix disabling logo in xml with `mapbox_logoEnabled` attribute.

### CODE REVIEW

File: plugin-attribution/src/test/java/com/mapbox/common/ShadowLogger.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import org.robolectric.annotation.Implementation;
+import org.robolectric.annotation.Implements;
+
+@Implements(Logger.class)
+public class ShadowLogger {
```
Comment (Chaoba) [inline plugin-attribution/src/test/java/com/mapbox/common/ShadowLogger.java:12]: We have many ShadowLogger.java distributed in several modules, maybe we could keep it in a test module or something else?

File: sdk/src/main/java/com/mapbox/maps/MapController.kt [hunk 2]
```diff
@@ -265,9 +265,10 @@ internal class MapController : MapPluginProviderDelegate, MapControllable {
       val logoPluginClass = Class.forName(PLUGIN_LOGO_CLASS_NAME) as Class<LogoPlugin>
       createPlugin(mapView, logoPluginClass)
     } catch (ex: ClassNotFoundException) {
-      Logger.d(
+      Logger.w(
         TAG,
-        "The Mapbox logo wordmark must remain enabled in accordance with our Terms of Service. See https://www.mapbox.com/legal/tos for more details."
+        mapView?.context?.getString(R.string.mapbox_warning_logo_disabled)
```
Comment (Chaoba) [inline sdk/src/main/java/com/mapbox/maps/MapController.kt:270]: We can use the context instance in `mapInitOptions` here, it's not nullable.

File: plugin-attribution/src/test/java/com/mapbox/common/ShadowLogger.java (hunk 1 above)
Comment (kiryldz) [reply plugin-attribution/src/test/java/com/mapbox/common/ShadowLogger.java:12]: I remember trying doing that when logger was first introduced but it appeared to be also not that easy to implement. Perhaps will put some time into it as operational excellence but not now.

File: sdk/src/main/java/com/mapbox/maps/MapController.kt (hunk 2 above)
Comment (kiryldz) [reply sdk/src/main/java/com/mapbox/maps/MapController.kt]: Very nice, thanks!

**Review by Chaoba — COMMENTED** [review]
LGTM

**Review by Chaoba — APPROVED** [review]

**Pull request merged**